src/plugin/context.rs

```diff
@@ -447,18 +447,16 @@ impl RapierContext {
                 );
 
                 if options.apply_impulse_to_dynamic_bodies {
-                    for collision in &*collisions {
-                        controller.solve_character_collision_impulses(
-                            dt,
-                            bodies,
-                            colliders,
-                            query_pipeline,
-                            (&scaled_shape).into(),
-                            shape_mass,
-                            collision,
-                            filter,
-                        )
-                    }
+                    controller.solve_character_collision_impulses(
+                        dt,
+                        bodies,
+                        colliders,
+                        query_pipeline,
+                        (&scaled_shape).into(),
+                        shape_mass,
+                        &*collisions,
```

Comment:
This is dependent on https://github.com/dimforge/rapier/pull/652

Reply:
migrarted to copy with [f582b34](https://github.com/dimforge/bevy_rapier/pull/525/commits/f582b342c72323df4a456fefe59a647fa10f9340)